### ISSUE TITLE
Following program causes garbage collection

### ISSUE THREAD

**Issue by petdr** [Abc-Arbitrage/ZeroLog#60]
````

using ZeroLog;
using ZeroLog.Appenders;
using ZeroLog.Configuration;

LogManager.Initialize(new ZeroLogConfiguration()
{
    LogMessagePoolSize = 10240,
    RootLogger =
    {
        Appenders =
        {
            new ConsoleAppender()
        }
    }
});

var logger = LogManager.GetLogger("Main");

var rand = new Random();
while (true)
{
    var r = rand.Next();
    Thread.Sleep(100);
    logger.Info($"rand: {r}");
}
````

when running the above program using 2.0.1 nuget package of ZeroLog, I can see the system is doing garbage collection.

Maybe I'm misunderstanding what is going on, but I thought that the logging was meant to be zero allocation and thus no garbage should be created.

**Comment by alexandrekow** [comment]
Thanks for pointing this out @petdr.

@ltrzesniewski I did a memory profiling and the allocations are coming from the `WriteMessage` method of the `StreamAppender`. 
Calling the `_encoding.GetBytes(chars, _byteBuffer)` method allocates `Byte[]` and `Char[]`

**Comment by MendelMonteiro** [comment]
So this is linked to the fact that we use `Console.OutputEncoding` as the encoding in the `ConsoleAppender`.  If I change it to be UTF8 it will no longer allocate.

@petdr , if you really don't want to have allocations when using the `ConsoleAppender` you could trivially copy and paste the existing implementation and hard-code the use of the UTF8 encoding as such:

```
    private Encoding _encoding = Encoding.UTF8;

    /// <summary>
    /// Initializes a new instance of the console appender.
    /// </summary>
    public ConsoleAppender()
    {
        Stream = Console.OpenStandardOutput();
        Encoding = _encoding;
        ColorOutput = !Console.IsOutputRedirected;
    }
```

Otherwise, it might be an idea on our side to add a flag to the `ConsoleAppender` to use the UTF8 encoding or the `OutputEncoding`.

**Comment by petdr** [comment]
Thanks for the prompt response and can confirm that in my real app I'm no longer seeing allocations.

**Comment by ltrzesniewski** [comment]
Well, oops. Sorry about that. 😬 

Turns out, `UTF8Encoding` has an optimization that `OSEncoding` does not: the base `Encoding` class allocates by default, and `UTF8Encoding` overrides that part. 😞  Maybe that's something I should report as an issue with `OSEncoding`...

Just FYI, you have two easier solutions that avoid allocations than copying the `ConsoleAppender` code:

- An appender that inherits `ConsoleAppender`
	```csharp
	class MyAppender : ConsoleAppender
	{
	    public MyAppender()
	    {
	        Encoding = Encoding.Default;
	    }
	}
	```

- Using `new TextWriterAppender(Console.Out)` with ZeroLog v2.1.0-pre2

In any case, that will be fixed in ZeroLog v2.1.0 final. 🙂

**Comment by ltrzesniewski** [comment]
Fixed in v2.1.0-pre3
